core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -432,26 +432,23 @@ private void checkTempBlockOwnedBySession(long sessionId, long blockId)
    */
   private void abortBlockInternal(long sessionId, long blockId) throws BlockDoesNotExistException,
       BlockAlreadyExistsException, InvalidWorkerStateException, IOException {
-    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE);
```

Comment:
@aaudiber @uronce-cc 
I think this is actually fine. here is my reason: this is a temp block, therefore no readers to compete lock. The only lock contention will happen among multiple writers on the same block (e.g., speculative execution). For this case, if we remove the block lock guard, exception will be thrown in `checkTempBlockOwnedBySession(sessionId, blockId)` as `tempBlockMeta` can not be found; if we keep the block lock, the same thing will happen. 

I guess this is fine only because this will always be creater/creater contention, which is a special case. @aaudiber , could you javadoc this somewhere?